students/BrianB/lesson07/html_render.py

```diff
@@ -0,0 +1,225 @@
+        out_file.write("<{}>".format(self.tag))
+        out_file.write(self.contents[0])
+        out_file.write("</{}>\n".format(self.tag))
+
+    def append(self, content):
+        raise NotImplementedError
+
+
+class H(OneLineTag):
+    """
+    I was unable to follow the instructions beyond
+    mid-Step 4, this was borrowed from an alt source
+    """
+    tag = 'h'
+
+    def __init__(self, level, content=None, **kwargs):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        try:
+            int(level)
+        except ValueError:
+            raise ValueError
+
+        self.tag = f'h{level}'
+        super().__init__(content, **kwargs)
+
+
+class Title(OneLineTag):
+    tag = "title"
+
+
+class A(OneLineTag):
+    """
+    I was unable to follow the instructions beyond
+    mid-Step 4, this was borrowed from an alt source
+    """
+    tag = 'a'
+
+    def __init__(self, link, content=None, **kwargs):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        if not (content and link): raise TypeError
+
+        kwargs['href'] = link
+        super().__init__(content, **kwargs)
+
+
+class SelfClosingTag(Element):
+    """
+    I was unable to follow the instructions beyond
+    mid-Step 4, this was borrowed from an alt source
+    """
+    def __init__(self, content=None, **kwargs):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        if content: raise TypeError
+        self.attributes = kwargs
+
+    def append(self, new_content):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        raise NotImplementedError
+
+    def render(self, out_file, indent="", ind_count=0):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        self.indent = indent * ind_count
+
+        out_file.write(f'{self.indent}{self._open_tag()[:-1]}')
+        for key, value in self.attributes.items():
+            out_file.write(f' {key}="{value}"')
+        out_file.write(f' />\n')
+
+
+class Hr(SelfClosingTag):
+    """
+    I was unable to follow the instructions beyond
+    mid-Step 4, this was borrowed from an alt source
+    """
+    tag = 'hr'
+
+
+# <br /> is XHTML format
+class Br(SelfClosingTag):
+    """
+    I was unable to follow the instructions beyond
+    mid-Step 4, this was borrowed from an alt source
+    """
+    tag = 'br'
+
+    def __init__(self, content=None, **kwargs):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        if kwargs: raise TypeError
```

Comment:
raise an Exception if there are kwargs, then pass them on? no need -- if they are sure to be empty then you don't need to pass them along.